compiler/z/codegen/OMRCodeGenerator.cpp

```diff
@@ -4732,10 +4730,23 @@ bool OMR::Z::CodeGenerator::getSupportsOpCodeForAutoSIMD(TR::ILOpCode opcode, TR
       return false;
       }
 
+   if (opcode.isVectorOpCode() && opcode.getVectorOperation() == OMR::vadd)
```

Comment:
I am thinking on how we make this part more extensible? i.e, when we start adding support for other vector operations as well. If the opcode is Vector Opcode, it might be better if we switch case similar to how it is used for old vector types. That would make adding new types easier and cleaner.

Reply:
When we remove all old vector opcodes we will have only one switch. Maybe better to leave it till then?

Reply:
And if` (opcode.isVectorOpCode())` will soon be gone as well.

Reply:
Are we planning to replace old opcode with new opcodes together in  future or that incrementally? 
If we are planning together then we can leave it till then but in case of incrementally adding them (For example, `OMR::vsub`, `OMR::vmul` , etc), we will have to stick them to above `if` condition right?

Reply:
Actually, just looked at it more careful. It  will not cause any nested switches, so sure, I will change to a single case switch to set a good example :)

Reply:
Changed to a switch on all platforms.